#!/usr/bin/env python
PACKAGE = "flir_gige"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# name / type / level / description / default / min / max
# Flir Ax5 has a maximum resolution of 320 x 256
gen.add("circles_per_row", int_t, 0,
        "circles per row", 4, 1, 20)
gen.add("circles_per_col", int_t, 0,
        "circles per col", 5, 1, 20)
gen.add("sigma", double_t, 0,
        "gaussian sigma", 1, 1, 20)
gen.add("min_area", double_t, 0,
        "contour min area", 40, 1, 1000)
gen.add("max_area", double_t, 0,
        "contour max area", 200, 100, 10000)
gen.add("thresh_window", int_t, 0,
        "adaptive threshold window size", 11, 3, 100)
gen.add("erosion_size", int_t, 0,
        "erosion size", 1, 1, 10)
size_enum = gen.enum([gen.const("mean", int_t, 0, "Adaptive thresh mean"),
                      gen.const("gaussian", int_t, 1, "Adaptive thresh gaussian")],
                     "Adaptive thresh type ")
gen.add("thresh_type", int_t, 0,
        "Adaptive thresh type", 0, 0, 1, edit_method=size_enum)

exit(gen.generate(PACKAGE, "flir_gige", "CalibProcDyn"))
